runtime/doc/api.txt

```diff
@@ -9,8 +9,7 @@ Nvim API							   *API* *api*
 Nvim exposes a powerful API that can be used by plugins and external processes
 via |msgpack-rpc|, Lua and VimL (|eval-api|).
 
-Nvim can also be embedded in C applications as libnvim, so the application
-can control the embedded instance by calling the C API directly.
+Applications can also embed libnvim to control via the C ABI.
```

Comment:
One should still use "C API", "ABI" is something defined by the system platform. But one could say that the "C API is ABI-stable" if we promise to not change types (`Dictionary` etc) defined in public headers. (But as we haven't properly separated libnvim public headers from internal headers yet, I don't think we should say that)